### PR TITLE
tests: improve and add mixerclient tests

### PR DESCRIPTION
**Pull request by kyessenov**
- upgrade to v2 xDS
- use gogo/proto, not golang/proto
- use proper envoy hot restart
- exit envoy gracefully, check for error code
- remove v1 tests
- enable TCP tests back, relax some conditions for less flakiness
- speed up stats checking
- add route directive test
- add destination uid override test
- add LDS change test for mixerfilter
- use stats for envoy readiness instead of pinging

/assign @lei-tang @JimmyCYJ @douglas-reid @hklai 

fixes https://github.com/istio/proxy/issues/1802
fixes https://github.com/istio/istio/issues/5696
fixes https://github.com/istio/istio/issues/6617
fixes https://github.com/istio/istio/issues/6265

### CODE REVIEW

**Comment by kyessenov** [comment]
Hey @yangminzhu , can you take a look here since you're familiar with envoy. Don't want this to become stale and run into merge conflicts.

**Comment by yangminzhu** [comment]
@kyessenov , I'm not familiar with envoy actually, maybe you mean @lizan ?

File: mixer/test/client/tcp_filter/tcp_filter_test.go [hunk 1]
```diff
@@ -66,13 +68,14 @@ const reportAttributesFailPost = `
   "context.protocol": "tcp",
   "context.time": "*",
   "mesh1.ip": "[1 1 1 1]",
+	"source.ip": "*",
```
Comment (JimmyCYJ) [inline mixer/test/client/tcp_filter/tcp_filter_test.go]: Please fix indent, it is not tab but two spaces.

File: mixer/test/client/tcp_filter/tcp_filter_test.go [hunk 2]
```diff
@@ -43,17 +44,18 @@ const reportAttributesOkPost = `
   "context.protocol": "tcp",
   "context.time": "*",
   "mesh1.ip": "[1 1 1 1]",
+	"source.ip": "[127 0 0 1]",
```
Comment (JimmyCYJ) [inline mixer/test/client/tcp_filter/tcp_filter_test.go]: Please fix indent.

Comment (kyessenov) [reply mixer/test/client/tcp_filter/tcp_filter_test.go]: done

File: mixer/test/client/disable_tcp_check_calls/disable_tcp_check_calls_test.go [hunk 3]
```diff
@@ -27,15 +27,16 @@ const reportAttributesOkPost = `
   "context.protocol": "tcp",
   "context.time": "*",
   "mesh1.ip": "[1 1 1 1]",
+	"source.ip": "[127 0 0 1]",
```
[the remark this answers is not in the document]
Comment (kyessenov) [reply mixer/test/client/disable_tcp_check_calls/disable_tcp_check_calls_test.go]: done

**Comment by kyessenov** [comment]
Thanks! I've noticed that I missed one tab. I'll patch it up later, the tests are very flaky lately for some reason.

**Comment by kyessenov** [comment]
Need re-lgtm. Connection may get refused on circle even when envoy reports readiness... Forced 1s delay to deal with this.

**Comment by kyessenov** [comment]
Thanks. Once again, vendor change created a merge conflict...

**Comment by hklai** [comment]
/lgtm

to unblock this. but please work with the CF test only to see what it takes to restore "--v2-config-only" (assuming that is necessary).

**Comment by kyessenov** [comment]
@hklai the flag is not necessary but cannot be specified twice. The original test was not meant for v2 or pilot at all, so it's a bit surprising why CF test got affected.

**Pull request merged**